android/src/main/java/io/customer/reactnative/sdk/CustomerIOReactNativeModule.kt

```diff
@@ -35,49 +41,54 @@ class CustomerIOReactNativeModule(
-
-        if (sdkInstance != null) {
-            logger.info("Customer.io instance already initialized, reinitializing")
-        }
-
-        val env = environment.toMap()
-        val config = configuration?.toMap()
-        val packageConfig = packageConfiguration?.toMap()
+        configJson: ReadableMap,
+        logLevel: String) {
+        val packageConfig = configJson.toMap()
+        val cdpApiKey = packageConfig[Keys.Config.CDP_API_KEY]
 
         try {
-            val newInstance = CustomerIOReactNativeInstance.initialize(
-                context = reactApplicationContext,
-                environment = env,
-                configuration = config,
-                packageConfig = packageConfig,
-                inAppEventListener = inAppMessagingModule,
+            val builder = CustomerIOBuilder(
+                applicationContext = reactApplicationContext.applicationContext as Application,
+                cdpApiKey = cdpApiKey.toString()
             )
-            logger.info("Customer.io instance initialized successfully from app")
-            // Request lifecycle events for first initialization only as relaunching app
-            // in wrapper SDKs may result in reinitialization of SDK and lifecycle listener
-            // will already be attached in this case as they are registered to application object.
-            if (!isLifecycleCallbacksRegistered) {
-                currentActivity?.let { activity ->
-                    logger.info("Requesting delayed activity lifecycle events")
-                    val lifecycleCallbacks = newInstance.diGraph.activityLifecycleCallbacks
-                    lifecycleCallbacks.postDelayedEventsForNonNativeActivity(activity)
-                }
+
+            // Auto track device attributes
+            (packageConfig[Keys.Config.AUTO_TRACK_DEVICE_ATTRIBUTES] as? Boolean)?.let {
+                builder.autoTrackDeviceAttributes(it)
+            }
+            // Migration site id for migration
+            (packageConfig[Keys.Config.MIGRATION_SITE_ID] as? String)?.let {
+                builder.migrationSiteId(it)
             }
-        } catch (ex: Exception) {
+            // Set region for workspace
+            (packageConfig[Keys.Config.REGION] as? String)?.let {
+                builder.region(Region.getRegion(it))
+            }
+            // Set log level
+            builder.logLevel(CioLogLevel.getLogLevel(logLevel))
+
+            // Flush at
+            (packageConfig[Keys.Config.FLUSH_AT] as? Int )?.let {
+                builder.flushAt(it)
+            }
+
+            // Flush at
+            (packageConfig[Keys.Config.FLUSH_INTERVAL] as? Int )?.let {
+                builder.flushInterval(it)
+            }
+
+            // To track application lifecycle events
+            (packageConfig[Keys.Config.TRACK_APP_LIFECYCLE_EVENTS] as? Boolean )?.let {
+                builder.trackApplicationLifecycleEvents(it)
+            }
```

Comment:
```suggestion
val builder = CustomerIOBuilder(
            applicationContext = reactApplicationContext.applicationContext as Application,
            cdpApiKey = cdpApiKey
        ).apply {
            packageConfig[Keys.Config.AUTO_TRACK_DEVICE_ATTRIBUTES] as? Boolean
                ?.let { autoTrackDeviceAttributes(it) }

            packageConfig[Keys.Config.MIGRATION_SITE_ID] as? String
                ?.let { migrationSiteId(it) }

            packageConfig[Keys.Config.REGION] as? String
                ?.let { region(Region.getRegion(it)) }

            logLevel(CioLogLevel.getLogLevel(logLevel))

            packageConfig[Keys.Config.FLUSH_AT] as? Int
                ?.let { flushAt(it) }

            packageConfig[Keys.Config.FLUSH_INTERVAL] as? Int
                ?.let { flushInterval(it) }

            packageConfig[Keys.Config.TRACK_APP_LIFECYCLE_EVENTS] as? Boolean
                ?.let { trackApplicationLifecycleEvents(it) }

            // TODO: Implement pushClickBehaviorAndroid when initializing messagingModule
        }
```

this could be simplified

Reply:
Thank you for the suggestion. ✅